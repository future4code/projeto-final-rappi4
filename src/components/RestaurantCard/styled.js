import styled from 'styled-components'

export const StyledRestaurantCard = styled.section`
height: 188px;
border-radius: 8px;
border: solid 1px #b8b8b8;
display: flex;
flex-direction: column;
`

export const StyledRestaurantImageWrapper = styled.div`
max-height: 60%;
border-radius: 8px 8px 0 0;
`

export const StyledRestaurantImage = styled.img`
height: 100%;
width: 100%;
object-fit: cover;
border-radius: 8px 8px 0 0;
`

export const StyledRestaurantTitle = styled.span`
display: block;
color: #e86e5a;
font-size: 16px;
margin-bottom: 6px;
`

export const StyledRestaurantContent = styled.div`
padding: 16px;
`

export const StyledRestaurantInfo = styled.div`
height: 18px;
font-size: 16px;
color: #b8b8b8;
display: flex;
justify-content: space-between;
`
