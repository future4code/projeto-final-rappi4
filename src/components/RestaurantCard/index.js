import React from 'react'
import { StyledRestaurantCard, StyledRestaurantImageWrapper, StyledRestaurantImage, StyledRestaurantContent, StyledRestaurantTitle, StyledRestaurantInfo } from './styled'

export default function RestaurantCard(props) {
    return (
        <StyledRestaurantCard>
            <StyledRestaurantImageWrapper>
                <StyledRestaurantImage src={props.restaurant.logoUrl} />
            </StyledRestaurantImageWrapper>
            <StyledRestaurantContent>
                <StyledRestaurantTitle>{props.restaurant.name}</StyledRestaurantTitle>
                <StyledRestaurantInfo>
                    <span>{props.restaurant.deliveryTime} min</span>
                    <span>Frete R$ {props.restaurant.shipping},00</span>
                </StyledRestaurantInfo>
            </StyledRestaurantContent>
        </StyledRestaurantCard>
    )
}
