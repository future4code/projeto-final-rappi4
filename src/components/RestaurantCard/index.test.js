import React from 'react';
import renderer from 'react-test-renderer';
import RestaurantCard from '.';
const mockRestaurant = {
    "id": "1",
    "description": "Habib's é uma rede de restaurantes de comida rápida brasileira especializada em culinária árabe, os restaurantes vendem mais de 600 milhões de esfirras por ano. A empresa emprega 22 mil colaboradores e tem 421 unidades distribuídas em mais de cem municípios em 20 unidades federativas.",
    "shipping": 6,
    "address": "Rua das Margaridas, 110 - Jardim das Flores",
    "name": "Habibs",
    "logoUrl": "http://soter.ninja/futureFoods/logos/habibs.jpg",
    "deliveryTime": 60,
    "category": "Árabe"
}
describe("Product Card",()=>{
    it("should render ",()=>{
        const tree = renderer.create(<RestaurantCard restaurant={mockRestaurant}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})