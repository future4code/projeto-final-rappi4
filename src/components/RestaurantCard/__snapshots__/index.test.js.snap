// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Card should render  1`] = `
<section
  className="sc-bdVaJa YuCSh"
>
  <div
    className="sc-bwzfXH bcpEuf"
  >
    <img
      className="sc-htpNat bRmsBL"
      src="http://soter.ninja/futureFoods/logos/habibs.jpg"
    />
  </div>
  <div
    className="sc-ifAKCX djZecA"
  >
    <span
      className="sc-bxivhb ibrqBp"
    >
      Habibs
    </span>
    <div
      className="sc-EHOje kbWHUg"
    >
      <span>
        60
         min
      </span>
      <span>
        Frete R$ 
        6
        ,00
      </span>
    </div>
  </div>
</section>
`;
